pangeo_forge_runner/commands/bake.py

```diff
@@ -266,10 +266,13 @@ def start(self):
                 # Set argv explicitly to empty so Apache Beam doesn't try to parse the commandline
                 # for pipeline options - we have traitlets doing that for us.
                 pipeline = Pipeline(options=pipeline_options, argv=[])
+
                 # Chain our recipe to the pipeline. This mutates the `pipeline` object!
-                # We expect `recipe` to either be a beam PTransform, or an object with a 'to_beam'
-                # method that returns a transform.
-                pipeline | recipe
+                # We expect `recipe` to be 1) a beam PTransform or 2) or a a string that leverages the
+                # `dict_object:` see `tests/test-data/gpcp-from-gcs/feedstock-0.10.x-dictobj/meta.yaml`
+                # as an example
+                if isinstance(recipe, PTransform):
```

Comment:
note to reviewer: zero idea how unit tests for `dict-object` are passing on `main` b/c we need the conditional here or they fail like they do on this PR 🤷

Reply:
another question @cisaacstern: we have a lot of special logic for `dict_object` recipes where we seem to not test [list recipes](https://github.com/pangeo-forge/pangeo-forge-runner/blob/main/tests/unit/test-recipes/list-recipes/feedstock/meta.yaml). Is the list version deprecated in favor of the dict version?